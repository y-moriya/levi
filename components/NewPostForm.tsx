import * as React from 'react'
import { useMutation, useQueryClient } from 'react-query'

const NewPostForm = () => {
  const queryClient = useQueryClient()

  const [form, update] = React.useState({
    content: '',
  })

  const { mutate } = useMutation(
    () => {
      return fetch('/api/posts', {
        method: 'POST',
        body: JSON.stringify(form),
      })
    },
    {
      onSuccess: () => {
        queryClient.invalidateQueries('posts')
      },
    }
  )

  const savePost = (ev: React.FormEvent<HTMLFormElement>) => {
    ev.preventDefault()

    update({ content: '' })

    mutate()
  }

  return (
    <div className="flex mx-auto items-center justify-center shadow-lg mt-8 mb-4 max-w-lg">
      <form
        onSubmit={savePost}
        className="w-full max-w-xl bg-white rounded-lg px-4 pt-2"
      >
        <div className="flex flex-wrap -mx-3 mb-6">
          <h2 className="px-4 pt-3 pb-2 text-gray-800 text-lg">
            Add a new Comment
          </h2>
          <div className="w-full md-w-full px-3 mb-2 mt-2">
            <textarea
              id="content"
              value={form.content}
              onChange={(e) => update({ ...form, content: e.target.value })}
              className="bg-gray-100 rounded border border-gray-400 leading-normal resize-none w-full h-20 py-2 px-3 font-medium placeholder-gray-700 focus:outline-none focus:bg-white"
              name="body"
              placeholder="Type Your Comment"
              required
            ></textarea>
          </div>
          <div className="w-full md-w-full flex items-start md:w-full px-3">
            <div className="mr-auto"></div>
            <div className="-mr-1">
              <button className="bg-white text-gray-700 font-medium py-1 px-4 border border-gray-400 rounded-lg tracking-wide mr-1 hover:bg-gray-100">
                Post
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  )
}

export default NewPostForm
