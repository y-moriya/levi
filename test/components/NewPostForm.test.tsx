import React from 'react'
import { render } from '../testUtils'
import { QueryClient, QueryClientProvider } from 'react-query'
import NewPostForm from '../../components/NewPostForm'

describe('New Post Form', () => {
  it('matches snapshot', async () => {
    const queryClient = new QueryClient({
      defaultOptions: {
        queries: {},
        mutations: {},
      },
    })

    const { asFragment } = render(
      <QueryClientProvider client={queryClient}>
        <NewPostForm />
      </QueryClientProvider>,
      {}
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
