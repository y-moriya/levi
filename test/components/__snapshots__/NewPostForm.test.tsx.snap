// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Post Form matches snapshot 1`] = `
<DocumentFragment>
  <form
    style="display: flex; flex-direction: column;"
  >
    <label
      style="display: flex; flex-direction: column;"
    >
      内容
      <input
        id="content"
        style="width: 500px;"
        type="text"
        value=""
      />
    </label>
    <button
      style="width: 100px; margin-top: 16px;"
    >
      Post
    </button>
  </form>
</DocumentFragment>
`;
